pkg/routeagent_driver/handlers/ovn/connection.go

```diff
@@ -186,45 +182,70 @@ func getTLSConfig(k8sClientset clientset.Interface) (*tls.Config, error) {
 	return tlsConfig, nil
 }
 
+/*
+   The discovery method is different for each kind of deployment.
+        kind non-ic: Get the db pod with label name=ovnkube-db, if present will get the protocol and other details from it
+
+        openshift non-ic: in this case there will be no pod with name=ovnkube-db but there will be a ovnkube-db service,
+        in this case we will return the default Openshift db path
+
+        kind ic with multiple node per zone: we parse every endpoint in the namespace ovn uses and find the one that has
+        same zone as the node in which route-agent runs.
+
+        kind ic with one node per zone: we will have no endpoints matching hence we will return default kind socket path
+
+        openshift ic with one node per zone: we will have endpoints in the namespace ovn but none will match the zone as
+        these are not db endpoints. Here we will return default Openshift socket path
+*/
+
 func discoverOvnKubernetesNetwork(ctx context.Context, k8sClientset clientset.Interface, zoneName string) (string, error) {
 	ovnDBPod, err := FindPod(ctx, k8sClientset, "name=ovnkube-db")
 	if err != nil {
 		return "", err
 	}
 
+	ovnPod, err := FindPod(ctx, k8sClientset, "app=ovnkube-node")
```

Comment:
In the description above, no where we talk about ovnkube-node, but this code is looking for this pod and if not found, its returning defaultOVNOpenshiftUnixSocket.

Reply:
it is to get the ovn namespace. This particular condition will hit in normal cases. But I just thought would return this value as the CNI is OVN. I changed the comment to include the pod.